packages/core/admin/ee/server/bootstrap.js

```diff
@@ -2,21 +2,50 @@
 
 // eslint-disable-next-line node/no-extraneous-require
 const { features } = require('@strapi/strapi/lib/utils/ee');
+const { mapAsync } = require('@strapi/utils');
 const executeCEBootstrap = require('../../server/bootstrap');
 const { getService } = require('../../server/utils');
 const actions = require('./config/admin-actions');
+const {
+  createReservedTable,
+  findTablesThatStartWithPrefix,
+  doesReservedTableEntryExist,
+  createReservedTableEntries,
+} = require('./utils/reserved-tables');
+
+module.exports = async ({ strapi }) => {
+  const connection = strapi.db.getSchemaConnection();
+  await createReservedTable(connection);
+
+  const tablesToPersist = [];
 
-module.exports = async () => {
   const { actionProvider } = getService('permission');
 
   if (features.isEnabled('sso')) {
     await actionProvider.registerMany(actions.sso);
   }
 
   if (features.isEnabled('audit-logs')) {
+    const auditLogTables = await findTablesThatStartWithPrefix('strapi_audit_logs');
+    tablesToPersist.push(...auditLogTables);
+
     await actionProvider.registerMany(actions.auditLogs);
   }
 
+  if (features.isEnabled('review-workflows')) {
+    const reviewWorkflowTables = await findTablesThatStartWithPrefix('strapi_workflows');
+    tablesToPersist.push(...reviewWorkflowTables);
+  }
```

Comment:
what if we did:
```suggestion
  if (features.isEnabled('review-workflows')) {
    await reserveTablesWithPrefix('strapi_workflows');
  }
```

We could have `reserveTablesWithPrefix` and `reserveTable` , and it could be something like:
```
const reserveTable = async (tableName) = {
  const reservedTables = await getReservedTables();
  
  // TODO Make sure it's not in the array first
  reservedTables.push(tableName)
  
    await strapi.store.set({
    type: 'core',
    key: 'reserved_tables',
    value: reservedTables,
  });
}
```

Side comment, do we care if the table does not exist 🤔 ?